redbot/core/core_commands.py

```diff
@@ -454,11 +464,89 @@ def get_bot_uptime(self, *, brief: bool = False):
             await ctx.send("No exception has occurred yet")
 
     @commands.command()
-    @checks.is_owner()
-    async def invite(self, ctx: commands.Context):
+    @commands.check(CoreLogic._can_get_invite_url)
+    async def invite(self, ctx):
         """Show's Red's invite url"""
         await ctx.author.send(await self._invite_url())
 
+    @commands.group()
+    @checks.is_owner()
+    async def inviteset(self, ctx):
+        """Setup the bot's invite"""
+        if not ctx.invoked_subcommand:
+            await ctx.send_help()
+
+    @inviteset.command()
+    async def public(self, ctx, confirm: bool = False):
+        """
+        Define if the command should be accessible\
+        for the average users.
+        """
+        if not self.bot.user.bot:
+            await ctx.send("I'm not a bot account. I have no invite URL.")
+            return
+        if await self.bot.db.invite_public():
+            await self.bot.db.invite_public.set(False)
+            await ctx.send("The invite is now private.")
+            return
+        app_info = await self.bot.application_info()
+        if not app_info.bot_public:
+            await ctx.send(
+                "I am not a public bot. That means that nobody except "
+                "you can invite me on new servers.\n\n"
+                "You can change this by ticking `Public bot` in "
+                "your token settings: "
+                "https://discordapp.com/developers/applications/me/{0}".format(self.bot.user.id)
+            )
+            return
+        if not confirm:
+            await ctx.send(
+                "You're about to make the `{0}invite` command public. "
+                "All users will be able to invite me on their server.\n\n"
+                "If you agree, you can type `{0}inviteset public yes`.".format(ctx.prefix)
+            )
+        else:
+            await self.bot.db.invite_public.set(True)
+            await ctx.send("The invite command is now public.")
+
+    @inviteset.command()
+    async def perms(self, ctx, level: int):
+        """
+        Make the bot create its own role with permissions on join.
+
+        The bot will create its own role with the desired permissions\
+        when he join a new server. This is a special role that can't be\
+        deleted or removed from the bot.
+
+        For that, you need to give a valid permissions level.
+        You can generate one [here](https://discordapi.com/permissions.html).
```

Comment:
I don't think this is appropriate to use embed specific formatting, as we specifically have a setting to disable embed use.